### ISSUE TITLE
Add support for TS 5.0 "extends" array

### ISSUE THREAD

**Issue by cspotcode** [TypeStrong/ts-node#1954]
TS 5.0 adds support for "extends" to be an array of one or more extended configs, not just a single config.

https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#supporting-multiple-configuration-files-in-extends

We can do the same, also merging `ts-node` options from multiple tsconfigs.

**Comment by evantill** [comment]
is it possible to add some warning that current released version does not support tsconfig extensions ?

**Comment by Kurt-von-Laven** [comment]
Thank you for addressing this issue! Would it be possible to release this fix when time allows?

**Comment by simenandre** [comment]
Let me know if I can help with anything in regard to releasing!

**Comment by 4lph4-Ph4un** [comment]
This is really needed! Any releases in the horizon?

**Comment by simenandre** [comment]
For anyone finding this issue and wanting a fix while we wait, plan, work or try to find time to work on a release.

It is possible to install `ts-node` from git, as @cspotcode noted in #1958.

There's a snippet and instructions here: https://github.com/TypeStrong/ts-node/discussions/1977

**Comment by BryceBarbara** [comment]
@simenandre Were you able to get that to work? It doesn't work for me because the repo is downloaded and there's no dist files.

**Comment by simenandre** [comment]
> @simenandre Were you able to get that to work? It doesn't work for me because the repo is downloaded and there's no dist files.

There are others having some issues with yarn, etc. Maybe you could repost your setup in the issue I referenced?

**Comment by tylim88** [comment]
still same issue with v10.9,2

**Comment by enisdenjo** [comment]
seriously? a year passed and this is still not released. please release a new version

**Comment by yogrr** [comment]
can be solved now by using

```
pnpm i -D ts-node@11.0.0-beta.1
```

worked for me with this version

don't forget to rm `node_modules` or rm old `ts-node` before

**Comment by rrooddooxx** [comment]
when is this going be released? is there something can do to help? thanks

**Comment by heyakyra** [comment]
In case anyone ends up here from Google, have been debugging various `jest` issues, and was hitting: 
```
Error: Jest: Failed to parse the TypeScript config file /var/home/[…]/storybook/jest.config.ts
  TypeError: value.replace is not a function
    at readConfigFileAndSetRootDir (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:2269:13)
    at async readInitialOptions (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:1147:13)
    at async readConfig (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:918:7)
    at async readConfigs (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:1168:26)
    at async runCLI (/var/home/[…]/storybook/node_modules/@jest/core/build/index.js:1393:7)
    at async Object.run (/var/home/[…]/storybook/node_modules/jest-cli/build/index.js:656:9)
```

and led here by https://github.com/jestjs/jest/issues/14096

but the above solution 

@yogrr on [on Jul 8, 2024](#issuecomment-2213552035)
> can be solved now by using
> 
> ```
> pnpm i -D ts-node@11.0.0-beta.1
> ```

just leads to another problem:

```
Error: Jest: Failed to parse the TypeScript config file /var/home/[…]/storybook/jest.config.ts
  TypeError: state.conditions.includes is not a function
    at readConfigFileAndSetRootDir (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:2269:13)
    at async readInitialOptions (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:1147:13)
    at async readConfig (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:918:7)
    at async readConfigs (/var/home/[…]/storybook/node_modules/jest-config/build/index.js:1168:26)
    at async runCLI (/var/home/[…]/storybook/node_modules/@jest/core/build/index.js:1393:7)
    at async Object.run (/var/home/[…]/storybook/node_modules/jest-cli/build/index.js:656:9)
```

 that was fixed in the more-recent stable version `10.9.2` so the beta version is older and does not include the fix 😖 

there is a patch available but a new beta release is definitely in order. https://github.com/TypeStrong/ts-node/issues/2076#issuecomment-2072081161
